src/ray/raylet/scheduling/cluster_task_manager.h

```diff
@@ -236,6 +239,11 @@ class ClusterTaskManager : public ClusterTaskManagerInterface {
   /// is still queued.
   std::unordered_map<SchedulingClass, std::deque<Work>> tasks_to_dispatch_;
 
+  /// Map of lease requests that waiting for workers popped.
+  /// Tasks move from dispatch to this map.
```

Comment:
Do we still need the dispatch queue if we have this queue? It seems that tasks won't spend any time in dispatch (can a pop ever fail now?)

Reply:
I have deleted this queue in latest code.